src/state_point.cpp

```diff
@@ -380,38 +380,42 @@ void load_state_point()
   // Read batch number to restart at
   read_dataset(file_id, "current_batch", simulation::restart_batch);
 
-  // Check for source in statepoint if needed
-  bool source_present;
-  read_attribute(file_id, "source_present", source_present);
-
   if (simulation::restart_batch > settings::n_batches) {
     fatal_error("The number batches specified in settings.xml is fewer "
       " than the number of batches in the given statepoint file.");
   }
 
+  // Logical flag for source present in statepoint file
+  bool source_present;
+  read_attribute(file_id, "source_present", source_present);
+
   // Read information specific to eigenvalue run
   if (settings::run_mode == RunMode::EIGENVALUE) {
     read_dataset(file_id, "n_inactive", temp);
     read_eigenvalue_hdf5(file_id);
 
     // Take maximum of statepoint n_inactive and input n_inactive
     settings::n_inactive = std::max(settings::n_inactive, temp);
+
+    // Check to make sure source bank is present
+    if (settings::path_sourcepoint == settings::path_statepoint &&
+        !source_present) {
+      fatal_error("Source bank must be contained in statepoint restart file");
+    }
   }
 
   // Read number of realizations for global tallies
   read_dataset(file_id, "n_realizations", simulation::n_realizations);
 
   // Set k_sum, keff, and current_batch based on whether restart file is part
   // of active cycle or inactive cycle
-  restart_set_keff();
-  simulation::current_batch = simulation::restart_batch;
-
-  // Check to make sure source bank is present
-  if (settings::path_sourcepoint == settings::path_statepoint &&
-      !source_present) {
-    fatal_error("Source bank must be contained in statepoint restart file");
+  if (settings::run_mode == RunMode::EIGENVALUE) {
+    restart_set_keff();
   }
 
+  // Set current batch number
+  simulation::current_batch = simulation::restart_batch;
```

Comment:
So it looks like the changes in this file are just lines of code getting moved around a bit -- can you explain the motivation for the change? Maybe it's just because I'm at the end of a long day, but it's not intuitively obvious to me :smile:

Reply:
The replacement of reading the source_present flag was undoing a previous change I made and places it in its original position. The conditional statement around the call to restart_set_keff() is to prevent this call when using a fixed source calculation as I found that this actually caused a segfault for fixed source restart cases as restart_set_keff() was calling variables which where not allocated.